### PR TITLE
ENH: Add min date function

### PR DESCRIPTION
**Pull request by bashtage**
Add min date function to match max date function

### CODE REVIEW

**Comment by bashtage** [comment]
I found this missing when attempting to merge from another datasource wanting to limit queries to match only the range of the tick data I have.  

There didn't seem to be any tests of `max_date`, so I haven't added any here.

**Comment by bmoscon** [comment]
@bashtage thanks! We do like unit/integration tests though, so please do add one for this. All new code should have associated tests.

File: arctic/tickstore/tickstore.py [hunk 1]
```diff
@@ -749,3 +749,16 @@ def max_date(self, symbol):
...
+        """
+        Return the minimum datetime stored for a particular symbol
+
+        Parameters
+        ----------
+        symbol : `str`
+            symbol name for the item
+        """
+        res = self._collection.find_one({SYMBOL: symbol}, projection={ID: 0, START: 1},
+                                        sort=[(START, pymongo.ASCENDING)])
```
Comment (dimosped) [inline arctic/tickstore/tickstore.py]: do we care about handling any of the below, or we prefer to let the exception bubble up?
 - res is None
 - START in res

maybe:
```
if not res:
     raise NoDataFoundException("No Data found for {}".format(symbol))
```

**Comment by bashtage** [comment]
That would be better than the status quo which raises KeyError

**Comment by bashtage** [comment]
LMK know if you think this is a good place for the test.

**Comment by bashtage** [comment]
@dimosped I have also added nicer errors when there are no min/max dates.

File: arctic/tickstore/tickstore.py (hunk 1 above)
Comment (bashtage) [reply arctic/tickstore/tickstore.py]: I put this in

**Pull request merged**